### PR TITLE
Additional documentation for development purpose

### PR DESCRIPTION
**Pull request by AlexeyPerevalov**
Issue: https://github.com/kubernetes-sigs/scheduler-plugins/issues/116

Signed-off-by: Alexey Perevalov <alexey.perevalov@huawei.com>

### CODE REVIEW

**Comment by AlexeyPerevalov** [comment]
While working on documentation for contributors, I found some aspects:
1. Debug symbols. I just described how to "enable" it, but it worth adding make target for debug binaries, and then describe how to attach into running scheduler.
2. I didn't include deployment specification neither for scheduler, nor for controller, but probably it could be interesting as well, especially for issues such that: https://github.com/kubernetes-sigs/scheduler-plugins/issues/127
3. README.md already has section Install, I didn't touch it, from my point of view it is incomplete, it just describes how to obtain images, but it doesn't tell what to do with it after.
4. Makefile doesn't **help** target as in k/k/Makefile, with built-in command description.

Point #1 and #4 - it's some kind of tasks and requires additional issue and PR. And maybe discussion, whether it's needed or not.
The 2 and 3 point could be addressed in this PR.

File: README.md [hunk 1]
```diff
@@ -41,6 +41,8 @@ You can reach the maintainers of this project at:
 - [Slack](https://kubernetes.slack.com/messages/sig-scheduling)
 - [Mailing List](https://groups.google.com/forum/#!forum/kubernetes-sig-scheduling)
 
+[Here](https://github.com/kubernetes-sigs/scheduler-plugins/blob/master/doc/develop.md) you can find an instruction how to build and run out-of-tree plugin.
```
Comment (damemi) [inline README.md:44]: more link text would be good, maybe change this to something like `You can find an [instruction how to build and run out-of-tree plugins here](https://github.com/kubernetes-sigs/scheduler-plugins/blob/master/doc/develop.md).`

File: doc/develop.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+command
+If you would like to build just the binaries you can do it by
+```shell
+make
+```
+
+## How to debug
+By default debug information is stripped, to keep it in the binary you need to remove -w command line option from
+ldflags in the Makefile. 
+
```
Comment (damemi) [inline doc/develop.md:29]: nit: extra newline

File: doc/develop.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# Developer guid
```
Comment (damemi) [inline doc/develop.md:1]: typo: `Developer guide`

**Review by damemi — COMMENTED** [review]
/ok-to-test

I think this is a great start for some of the frequent questions we see about how to use this repo. Maybe even add some more info about how this is an example custom scheduler built with the framework too?

**Comment by damemi** [comment]
@AlexeyPerevalov I think your points on documenting how to actually run the scheduler are a good idea too. There seems to be a lot of confusion on how to "use" these plugins that would be helpful to point people to

**Comment by AlexeyPerevalov** [comment]
> /ok-to-test
> 
> I think this is a great start for some of the frequent questions we see about how to use this repo. Maybe even add some more info about how this is an example custom scheduler built with the framework too?

Could you please clarify, do you mean to add some links to https://kubernetes.io/docs/concepts/scheduling-eviction/scheduling-framework/  or maybe a  link to keps/sig-scheduling/624-scheduling-framework/README.md?

**Comment by damemi** [comment]
@AlexeyPerevalov I was more referring to what you said in https://github.com/kubernetes-sigs/scheduler-plugins/pull/128#issuecomment-739748988 (I made my review comment before I saw your other comment, sorry). Those are also great ideas of things to link to though.

File: doc/develop.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+```shell
+make
+```
+
+## How to debug
+By default debug information is stripped, to keep it in the binary you need to remove -w command line option from
+ldflags in the Makefile.
+
+## How to start
+If you would like to start produced kube-scheduler image you can use it in your static kube-scheduler manifests or any kind of
```
Comment (Huang-Wei) [inline doc/develop.md:31]: Missing some words here?

File: doc/develop.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+```
+After that you'll take localhost:5000/scheduler-plugins/kube-scheduler:latest and
+localhost:5000/scheduler-plugins/controller:latest images.
+
+Your source code has to be located at `$GOPATH/src/sigs.k8s.io`, if you would like to change some type definition of the interfaces
+and regenerate their implementation by running
+```shell
+hack/update-codegen.sh
+```
+command
```
Comment (Huang-Wei) [inline doc/develop.md:20]:
```suggestion
command.
```

**Comment by Huang-Wei** [comment]
> The 2 and 3 point could be addressed in this PR.

Given that it's a developer guide instead of a user guide, this doc should put more focus on (1) how to run/debug it locally, and (2) probably add a section to introduce how to test (unit/integration test) locally, which should be helpful for developers.

We can compose other docs for users later, to guide users on how to use/configure the binary.

**Review by seanmalloy — APPROVED** [review]

**Comment by Huang-Wei** [comment]
> /assign @Huang-Wei

https://github.com/kubernetes-sigs/scheduler-plugins/pull/128#discussion_r539604107 is good to be resolved :)

File: doc/develop.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
+# Developer guide
+
+## How to build
+
+This section is about building on the local host, e.g. for debug purpose.
+Your `GOPATH` environment variable should be set.
```
[the remark this answers is not in the document]
Comment (seanmalloy) [reply doc/develop.md:6]: I way I read it this comment is resolved by https://github.com/kubernetes-sigs/scheduler-plugins/pull/128/files#diff-66e9efc4e747562f39d585f75eb79b480404a58111ccae618b8c98d09f36c3faR26-R30. But I could be wrong.

**Comment by seanmalloy** [comment]
> > /assign @Huang-Wei
> 
> [#128 (comment)](https://github.com/kubernetes-sigs/scheduler-plugins/pull/128#discussion_r539604107) is good to be resolved :)

The way I read it this comment is resolved by https://github.com/kubernetes-sigs/scheduler-plugins/pull/128/files#diff-66e9efc4e747562f39d585f75eb79b480404a58111ccae618b8c98d09f36c3faR26-R30. But I could be wrong.

**Comment by AlexeyPerevalov** [comment]
> 
> 
> > /assign @Huang-Wei
> 
> [#128 (comment)](https://github.com/kubernetes-sigs/scheduler-plugins/pull/128#discussion_r539604107) is good to be resolved :)

It was resolved on line 26 of develop.md

**Comment by Huang-Wei** [comment]
> It was resolved on line 26 of develop.md

Oh, I misread.

/approve

Thanks @AlexeyPerevalov !

**Pull request merged**